### PR TITLE
Minor rewording of 2.6 to make it easier to understand.

### PR DESCRIPTION
**Pull request by viktorklang**
Addressing #321 with minimal changes. (No changes in spec semantics)

@reactive-streams/contributors Please review and vote.

### CODE REVIEW

**Comment by ktoso** [comment]
Wanting to look at the changeset shows me a 500 error page; does it work for you?

**Comment by akarnokd** [comment]
@ktoso Same here. I had trouble with #339 as well the past week, mostly since the Amazon outage. The patch links from the email work for me:

https://github.com/reactive-streams/reactive-streams-jvm/pull/342.patch
https://github.com/reactive-streams/reactive-streams-jvm/pull/342.diff

**Comment by ktoso** [comment]
Thanks, will have a look

**Comment by viktorklang** [comment]
Yeah, I have issues with GH as well, 500s all over the place.

**Comment by ktoso** [comment]
Uhm, if the change is only to add the `,` then I don't think it's helping enough.
The linked issue proposes a more wordy version, however we could find a balance between the current and the new one perhaps?

How about:

> A `Subscriber` MUST call `Subscription.cancel()` when it is no longer interested (or able) in receiving signals from its `Publisher`, even without the `Publisher` having signaled `onError` or `onComplete`.

**Comment by viktorklang** [comment]
not, it changes "the Publisher" to "its Publisher" and adds the comma.

**Comment by ktoso** [comment]
I see, the `valid to its...` still is not completely clear though, as witnessed by that issue. We don't use that wording anywhere else either AFAICS. A subscription can be valid, but this is the single case we used it for a Subscriber.

**Comment by viktorklang** [comment]
No, it is not about the Subscriber being valid, it is about the Subscription not being valid.

**Comment by ktoso** [comment]
Proves my point that it's easy to confuse the "it" then though?

**Comment by viktorklang** [comment]
@ktoso Very good point. See my second attempt.

**Comment by ktoso** [comment]
Hmm, simple wording and I think it does indeed cover all the cases we mean here. 
The extended intent section explains what we mean well now. 

Sounds good to me 👍

File: README.md [hunk 1]
```diff
@@ -139,8 +139,8 @@ public interface Subscriber<T> {
...
 | <a name="2.5">5</a>       | A `Subscriber` MUST call `Subscription.cancel()` on the given `Subscription` after an `onSubscribe` signal if it already has an active `Subscription`. |
 | [:bulb:](#2.5 "2.5 explained") | *The intent of this rule is to prevent that two, or more, separate Publishers from thinking that they can interact with the same Subscriber. Enforcing this rule means that resource leaks are prevented since extra Subscriptions will be cancelled.* |
-| <a name="2.6">6</a>       | A `Subscriber` MUST call `Subscription.cancel()` if it is no longer valid to the `Publisher` without the `Publisher` having signaled `onError` or `onComplete`. |
-| [:bulb:](#2.6 "2.6 explained") | *The intent of this rule is to establish that Subscribers cannot just throw Subscriptions away when they are no longer needed, they have to call `cancel` so that resources held by that Subscription can be safely, and timely, reclaimed.* |
+| <a name="2.6">6</a>       | A `Subscriber` MUST call `Subscription.cancel()` if the `Subscription` is no longer needed. |
```
Comment (viktorklang) [inline README.md]: One question which comes to mind is if we need to be explicit here that this rule only applies as long as it doesn't violate any other rule (i.e. not calling Subscription methods after terminal state has been received etc.)

File: README.md [hunk 2]
```diff
@@ -139,8 +139,8 @@ public interface Subscriber<T> {
...
-| <a name="2.6">6</a>       | A `Subscriber` MUST call `Subscription.cancel()` if it is no longer valid to the `Publisher` without the `Publisher` having signaled `onError` or `onComplete`. |
-| [:bulb:](#2.6 "2.6 explained") | *The intent of this rule is to establish that Subscribers cannot just throw Subscriptions away when they are no longer needed, they have to call `cancel` so that resources held by that Subscription can be safely, and timely, reclaimed.* |
+| <a name="2.6">6</a>       | A `Subscriber` MUST call `Subscription.cancel()` if the `Subscription` is no longer needed. |
+| [:bulb:](#2.6 "2.6 explained") | *The intent of this rule is to establish that Subscribers cannot just throw Subscriptions away when they are no longer needed, they have to call `cancel` so that resources held by that Subscription can be safely, and timely, reclaimed. An example of this would be a Subscriber which is only interested in a specific element, which would then cancel its Subscription to signal its completion to the Publisher.* |
```
Comment (viktorklang) [inline README.md]: Is this example/clarification good enough?

**Comment by viktorklang** [comment]
@reactive-streams/contributors Yay or nay?

**Comment by viktorklang** [comment]
@reactive-streams/contributors Ping!

**Pull request merged**